### DISCUSSION TITLE
Detect bindings.gyp

### DISCUSSION THREAD

**Discussion by tbroyer** [naugtur/can-i-ignore-scripts#21] [Ideas]
According to [the docs](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#default-values), NPM will default to `"scripts": { "install": "node-gyp rebuild" } }` if there's a `bindings.gyp` file next to the `package.json` and there's no `preinstall` or `install` script explicitly defined.

It looks like can-i-ignore-scripts does not look for bindings.gyp files and thus could miss packages that would have to be rebuilt.

**Comment by naugtur** [comment]
I became aware of that soon after creating the package, but decided to wait for someone to report it or find some time to research it (I could basically detect all packages with implicit gyp builds and add them to the keep list. 

Are you interested in working on implicit gyp build detection?

**Reply by tbroyer** [reply]
Here we go: https://github.com/naugtur/can-i-ignore-scripts/pull/22

I added the _default script_ as if it was explicit in the `package.json`, i.e. it's shown in the output:
```shell-session
$ node ./index.js 

█▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀█
  ▄▄·  ▄▄▄·  ▐ ▄    ▄     ▪    ▄▄     ▐ ▄       ▄▄▄   ▄▄▄      ▄▄▄▄·
 ▐█ ▌ ▐█ ▀█ ·█▌▐█   ██    ██  ▐█ ▀    █▌▐█      ▐▄ █· █  ▀·  .▀  .█▌
 ██ ▄▄▄█▀▀█ ▐█▐▐▌   ▐█·   ▐█· ▄█ ▀█▄ ▐█▐▐▌ ▄█▀▄ ▐▀▀▄ ▐█▀      ▄█▀▀▀·
 ▐███▌▐█ ▪▐▌██▐█▌   ▐█▌   ▐█▌ ▐█▄ ▐█ ██▐█▌▐█▌.▐▌▐▄ █▌▐█▄▄▄▌   ▀
 ·▀▀▀  ▀  ▀ ▀▀ █▪   ▀▀▀   ▀▀▀ ·▀▀▀▀  ▀▀ █▪ ▀█▄▀▪.▀  ▀ ▀▀▀     ▀

▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
Found following packages with scripts:
[ ignore ] 'spawn-sync' has scripts but they can be ignored 
             reason: installs native modules for old node.js versions
[ check? ] 'test-please-ignore' needs reviewing (it uses gyp, so you probably need it)
[
  {
    name: 'test-please-ignore',
    version: '1.0.0',
    scripts: { install: 'node-gyp rebuild' },
    path: 'node_modules/test-please-ignore/'
  }
]
[  keep  ] 'pre-commit' needs its scripts to run 
             reason: used in local development, can skip for CI

▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
What now? Run rebuild after 'npm ci --ignore-scripts' to trigger 
scripts you need to keep. 

A suggestion to get you started:
npm rebuild pre-commit test-please-ignore
█▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄█

If you check some packages, consider contributing your findings on github.
```

If you'd prefer to keep it implicit, then I could add a boolean flag to the `found` records, and then check it in addition to looking for `gyp ` in the (explicit) install scripts.
